python/ray/data/preprocessors/normalizer.py

```diff
@@ -7,16 +7,69 @@
 
 
 class Normalizer(Preprocessor):
-    """Normalize each record to have unit norm.
+    r"""Scales each sample to have unit norm.
 
-    Supports the following normalization types:
-        * l1: Sum of the absolute values.
-        * l2: Square root of the sum of the squared values.
-        * max: Maximum value.
+    This preprocessor works by dividing each sample (i.e., row) by the sample's norm.
+    The general formula is given by
+
+    .. math::
+
+        s' = \frac{s}{\lVert s \rVert_p}
+
+    where :math:`s` is the sample, :math:`s'` is the transformed sample, and :math:`p`
+    is the norm type.
+
+    The following norms are supporting:
+
+        * :math:`L^1`: Sum of the absolute values.
+        * :math:`L^2`: Square root of the sum of the squared values.
+        * :math:`L^\infty`: Maximum value.
 
     Args:
-        columns: The columns that in combination define the record to normalize.
-        norm: "l1", "l2", or "max". Defaults to "l2"
+        columns: The columns to include in samples.
+        norm: The norm to use. The supported values are ``"l1"``, ``"l2"``, or
+            ``"max"``. Defaults to ``"l2"``.
+
+    Raises:
+        ValueError: if ``norm`` is not ``"l1"``, ``"l2"``, or ``"max"``.
+
+    Examples:
+        >>> import pandas as pd
+        >>> import ray
+        >>> from ray.data.preprocessors import Normalizer
+        >>>
+        >>> df = pd.DataFrame({"X1": [1, 1], "X2": [1, 0], "X3": [0, 1]})
+        >>> ds = ray.data.from_pandas(df)  # doctest: +SKIP
+        >>> ds.to_pandas()  # doctest: +SKIP
+           X1  X2  X3
+        0   1   1   0
+        1   1   0   1
+
+        The :math:`L^2`-norm of the first sample is :math:`\sqrt{2}`, and the
+        :math:`L^2`-norm of the second sample is :math:`1`.
```

Comment:
Don't know how useful these descriptions are. Could use feedback on how to make the example descriptions more useful

Reply:
Maybe add it directly as a comment in the code block?